### PR TITLE
Keg git branch reset task

### PR DESCRIPTION
**Pull request by lancetipton**
## Context

* Michael wanted a `git branch reset` task and said it would be nice to have if I was so inclined.
* I was!

## Goal

* Add a task to reset a git branch

## Updates

* `repos/keg-cli/src/libs/git/branch.js`
  * Added helpers to clean and reset git branches
* `repos/keg-cli/src/tasks/git/index.js`
  * Updated to work like most other root task files
  * Created a `git.js` file and moved code from `index.js` into that
  * This is just to keep things consistent. No actual code changes
* `repos/keg-cli/src/tasks/git/branch/branch.js` / `repos/keg-cli/src/tasks/git/git.js`
  * Added `reset` task to `git` and  `git branch` tasks
    * Both reference the same `reset` task file. 
    * I put it in branch, because it is a branch task
    * But it's also nice to have at the root level. Less typing 
* `repos/keg-cli/src/tasks/git/branch/reset.js`
  * Task to reset the current git branch

## Testing

**Setup**
* Check out this pull request with `keg && keg pr 27`
* Cd into tap or git repo

**Test 1**
* Make a change to a file
* Run the reset cmd => `keg reset`
  * Should ask you to confirm the reset task
    * Type `n`, and hit enter
  * Ensure the change to the repo was **NOT** reset
* Run the reset cmd `keg reset` again
  * Should ask you to confirm the reset task
    * Type `y`, and hit enter
  * Ensure the change to the repo was reset

**Test 2**
* Make a change to a file
* Run the reset cmd with the force option `keg reset -f`
  * Should **NOT** ask you to confirm the reset task
  * Ensure the change to the repo was reset

### CODE REVIEW

File: repos/keg-cli/src/tasks/git/branch/reset.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    alias: [ 'clean' ],
+    description: 'Resets a git branch, by removes all changes and untracked files',
+    action: branchReset,
+    options: {
+      context: {
+        description: 'Name of the git repo to reset',
+        example: 'keg git branch reset --context core',
+      },
+      tap: {
+        description: 'Name of the tap to build a Docker image for',
```
Comment (daniel112) [inline repos/keg-cli/src/tasks/git/branch/reset.js:54]: wrong description here

**Review by daniel112 — CHANGES_REQUESTED** [review]
works great! 
there's just 1 description that needs to be updated

File: repos/keg-cli/src/tasks/git/branch/reset.js (hunk 1 above)
Comment (lancetipton) [reply repos/keg-cli/src/tasks/git/branch/reset.js]: good catch - updated

**Review by daniel112 — APPROVED** [review]
looks good!

**Pull request merged**